src/WPF/WPF.Viewer/Samples/GraphicsOverlay/CreateAndEditGeometries/CreateAndEditGeometries.xaml.cs

```diff
@@ -0,0 +1,421 @@
+
+                _graphicsOverlay.Graphics.Add(graphic);
+            }
+
+            // Allow the creation of new graphics.
+            EnableGeometryButtons();
+            ToolComboBox.IsEnabled = true;
+        }
+
+        // Stops the geometry editor without saving the geometry stored within.
+        private void DiscardButton_Click(object sender, RoutedEventArgs e)
+        {
+            if (_selectedGraphic != null)
+            {
+                // Make visisble the pre-existing graphic when editing.
+                _selectedGraphic.IsVisible = true;
+                _selectedGraphic.IsSelected = false;
+                _selectedGraphic = null;
+
+                _geometryEditor.ClearSelection();
+            }
+
+            _geometryEditor.Stop();
+
+            // Allow for the creation of new graphics.
+            EnableGeometryButtons();
+            ToolComboBox.IsEnabled = true;
+        }
+
+        // Removes all graphics from the graphics overlay.
+        private void DeleteAllButton_Click(object sender, RoutedEventArgs e)
+        {
+            _graphicsOverlay.Graphics.Clear();
+
+            // Disable the delete all button as there are no graphics to delete.
+            DeleteAllButton.IsEnabled = false;
+            _geometryCreated = false;
+
+            // Allow for the creation of new graphics.
+            EnableGeometryButtons();
+            ToolComboBox.IsEnabled = true;
+        }
+
+        #endregion Edit controls event handlers
+
+        #region Enable and disable geometry buttons methods
+
+        // Ensure all geometry buttons are enabled.
+        private void EnableGeometryButtons()
+        {
+            foreach (var button in _geometryButtons)
+            {
+                button.IsEnabled = true;
+            }
+        }
+
+        // Disable all geometry buttons besides the one that was just clicked.
+        private void DisableOtherGeometryButtons(Button keepEnabled)
+        {
+            foreach (var button in _geometryButtons.Where(value => value != keepEnabled))
+            {
+                button.IsEnabled = false;
+            }
+        }
+
+        #endregion Enable and disable geometry buttons methods
+
+        private async void MyMapView_GeoViewTapped(object sender, GeoViewInputEventArgs e)
+        {
+            try
+            {
+                if (!_geometryEditor.IsStarted)
+                {
+                    // Identify graphics in the graphics overlay using the mouse point.
+                    IReadOnlyList<IdentifyGraphicsOverlayResult> results = await MyMapView.IdentifyGraphicsOverlaysAsync(e.Position, 2, false);
+
+                    // Get the first graphic if results were found.
+                    IdentifyGraphicsOverlayResult idResult = results.FirstOrDefault();
+                    if (idResult != null && idResult.Graphics.Count > 0)
+                    {
+                        _selectedGraphic = idResult.Graphics.FirstOrDefault();
```

Comment:
Since you do a prior check for count, I think this can use index `idResult.Graphics[0]` or if you want even simplify that
```CSharp
if (MyMapView.GeometryEditor.IsStarted) return; // editing already in progress

_selectedGraphic = results.FirstOrDefault()?.Graphics?.FirstOrDefault();

if(_selectedGraphic is null)  return; // nothing to edit
```